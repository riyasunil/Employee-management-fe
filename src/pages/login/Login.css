.login__container{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    /* border: 2px solid red; */
}
