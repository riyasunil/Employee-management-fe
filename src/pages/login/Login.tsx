import "./Login.css";
import LeftPanel from "../../components/leftpanel/LeftPanel";
import RightPanel from "../../components/rightpanel/RightPanel";
import { Navigate } from "react-router-dom";


const Login = () => {
  return (
    <>
      <div className="login__container">
        <LeftPanel />
        <RightPanel />
      </div>
    </>
  );
};

export default Login;
