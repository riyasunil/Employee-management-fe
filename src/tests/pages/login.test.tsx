import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import userEvent from "@testing-library/user-event";
import Login from "../../pages/login/Login";
import { Provider } from "react-redux";
// Mock react-router-dom's useNavigate
const mockNavigate = vi.fn();
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});
// Mock the store
const mockStore = configureStore({
  reducer: {
    employee: (state = { employees: [] }) => state,
    department: (state = { departments: [] }) => state,
    employeeApi: (state = {}) => state,
  },
});
// Mock the login mutation
const mockLoginMutation = vi.fn();
vi.mock("../../api-services/auth/login.api", () => ({
  useLoginMutation: () => [
    () => ({
      unwrap: () => mockLoginMutation(),
    }),
    { isLoading: false },
  ],
}));
describe("Login Page", () => {
  const renderLogin = () => {
    return render(
      <Provider store={mockStore}>
        <Login />
      </Provider>
    );
  };
  beforeEach(() => {
    vi.clearAllMocks();
  });
  it("should match snapshot", () => {
    const { container } = renderLogin();
    expect(container).toMatchSnapshot();
  });
  it("should render login form with all required elements", () => {
    renderLogin(); // Check if username input is present
    const usernameInput = screen.getByLabelText("Email");
    expect(usernameInput).toBeInTheDocument(); // Check if password input is present
    const passwordInput = screen.getByLabelText("Password");
    expect(passwordInput).toBeInTheDocument(); // Check if submit button is present
    const submitButton = screen.getByRole("button", { name: /login/i });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute("type", "button");
   
  });

   it("should handle successful login", async () => {
      mockLoginMutation.mockResolvedValueOnce({ token: "fake-token" });
      renderLogin(); // Fill in the form
      const usernameInput = screen.getByLabelText("Email");
      const passwordInput = screen.getByLabelText("Password");
      const submitButton = screen.getByRole("button", { name: /login/i });
      await userEvent.type(usernameInput, "test@example.com");
      await userEvent.type(passwordInput, "password123"); // Submit the form
      await userEvent.click(submitButton); // Wait for the login mutation to resolve and navigation to occur
      await waitFor(() => {
        expect(mockLoginMutation).toHaveBeenCalled();
        expect(mockNavigate).toHaveBeenCalledWith("/employee");
      });
    });
});
