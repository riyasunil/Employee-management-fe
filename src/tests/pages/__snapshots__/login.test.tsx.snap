// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match snapshot 1`] = `
<div>
  <div
    class="login__container"
  >
    <div
      class="leftpanel__container"
    >
      <div
        class="circular__mask"
      >
        <img
          alt="left panel"
          src="/src/assets/kv-login.jpeg"
        />
      </div>
    </div>
    <div
      class="rightpanel__container"
    >
      <div
        class="login_input"
      >
        <img
          alt=""
          src="/src/assets/kv-logo.png"
        />
        <div
          class="input_wrapper"
        >
          <input
            id="email"
            placeholder="Email"
            type="text"
            value=""
          />
          <label
            for="email"
          >
            Email
          </label>
        </div>
        <div
          class="input_wrapper"
        >
          <input
            id="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
          <div
            class="input__right__icon"
          >
            <button
              style="display: flex; padding: 2px; background-color: white; box-shadow: none; border: 1px solid gray; border-radius: 50%; width: 22px; height: 22px; align-items: center; justify-content: center; cursor: pointer;"
              type="button"
            >
              X
            </button>
          </div>
        </div>
        <div
          class="show_password"
        >
          <input
            type="checkbox"
          />
          Show password
        </div>
        <button
          class="btn"
          style="width: 100%; color: white; background-color: rgb(3, 174, 238);"
          type="button"
        >
           
          Login false
        </button>
      </div>
    </div>
  </div>
</div>
`;
